src/Cassandra/Requests/IRequest.cs

```diff
@@ -25,5 +25,11 @@ internal interface IRequest
         /// Writes the frame for this request on the provided stream
         /// </summary>
         int WriteFrame(short streamId, MemoryStream stream, ISerializer serializer);
+
+        /// <summary>
+        /// Result Metadata to parse the response rows. Only EXECUTE requests set this value so it will be null
+        /// for other types of requests.
+        /// </summary>
+        ResultMetadata ResultMetadata { get; }
```

Comment:
We should move this declaration to `IQueryRequest`, that way we avoid to create it for requests that don't make sense to have it.

Reply:
I thought about doing that but then I would have to add a `request as IQueryRequest` statement on the `OperationState` constructor. Do you think we should add that cast instead of defining this property on every request class?

Reply:
I was saying that because having the property in each implementation didn't make much sense for me (i.e., having a `ResultMetadata` prop within `OptionsRequest`), but it doesn't affect execution and its a matter of opinions so I understand if you prefer leaving it as is and avoid a cast.

Reply:
Oh I agree that it doesn't make sense, it's just that my personal opinion is that the cast is worse here and other potentially better alternatives would require rewriting a lot of stuff. 

I'll go ahead and merge as is but I'll keep this in mind in the future (maybe when implementing protocol checksumming if it requires a lot of changes to this code?)